s3/src/main/scala/akka/stream/alpakka/s3/model.scala

```diff
@@ -464,3 +464,17 @@ final class ObjectMetadata private (
 object ObjectMetadata {
   def apply(metadata: Seq[HttpHeader]) = new ObjectMetadata(metadata)
 }
+
+/**
+ * While checking for bucket access those responses are available
+ * 1) AccessDenied - User does have permission to perform ListBucket operation, so bucket exits
+ * 2) AccessGranted - User doesn't have rights to perform ListBucket but bucket exits
+ * 3) NotExists - Bucket doesn't exit
+ *
+ * @see https://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketHEAD.html
+ */
+sealed trait BucketAccess
+
+case object AccessDenied extends BucketAccess
```

Comment:
For Java-friendliness:
1. change `BucketAccess` from `trait` to `class`
2. Put the objects in `object BucketAccess`
3. Add values in the `BucketAccess` object (`val accessDenied: BucketAccess = AccessDenied`)